squeal-postgresql/src/Squeal/PostgreSQL/Expression/Text.hs

```diff
@@ -63,3 +70,19 @@ like = unsafeBinaryOp "LIKE"
 -- ((E'abc' :: text) ILIKE (E'a%' :: text))
 ilike :: Operator (null 'PGtext) (null 'PGtext) ('Null 'PGbool)
 ilike = unsafeBinaryOp "ILIKE"
+
+-- | Determines the location of the substring match using the `strpos`
+-- function. If `0` is returned, no match, else the index of the match
+-- location is returned.
+strpos
+  :: '[null0 'PGtext, null1 'PGtext] ---> (CombineNullity null0 null1) 'PGint4
+strpos = unsafeFunctionN "strpos"
+
+-- | Returns a `Null PGbool` indicating whether or not the second argument
+-- is contained within the first argument.
+hasInfix
+  :: forall grp lat with db params from null0 null1.
+     Expression grp lat with db params from (null0 'PGtext)
+  -> Expression grp lat with db params from (null1 'PGtext)
+  -> Expression grp lat with db params from ('Null 'PGbool)
```

Comment:
it's possible to change this type to `Operator (null0 'PGtext) (null1 'PGtext) ('Null 'PGbool)`

Reply:
Oh, that's much more concise, ty